src/imginputfileconn.h

```diff
@@ -287,8 +326,16 @@ namespace dd
     
     int feature_size() const
     {
-      if (_bw || _unchanged_data) return _width*_height; // XXX: only valid for single channels
-      else return _width*_height*3; // RGB
+      if (_bw || _unchanged_data) {
+		  // XXX: only valid for single channels
+      	if (_crop_width != 0 && _crop_height != 0) return _crop_width*_crop_height;
+      	else return _width*_height;
+      }
+      else {
+	  	// RGB
+	  	if (_crop_width != 0 || _crop_height != 0) return _crop_width*_crop_height*3;
```

Comment:
Any issue if either `crop_width` or `crop_height` is unset ?

Reply:
Yup, sorry that should be `&&` as on line 331. Thanks for catching that.